smithy-typescript-codegen-test/smithy-build.json

```diff
@@ -8,7 +8,8 @@
             "packageVersion": "0.0.1",
             "packageJson": {
                 "license": "Apache-2.0"
-            }
+            },
+            "compatibilityMode": "strict"
```

Comment:
Can this be made more specific? Glancing at smithy-build.json, I don't have a good idea what compatibility this is will impact.